src/main/java/com/example/animalshelter/model/Animal.java

```diff
@@ -1,25 +1,57 @@
 package com.example.animalshelter.model;
 
-import lombok.Builder;
 import lombok.Data;
 
-import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Positive;
+import javax.persistence.*;
 
 /**
  * Keeps basic information about an animal
  */
 @Data
-@Builder
+@Table
+@Entity
 public class Animal {
 
-    @NotNull
-    @Positive
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column
     private Long id;
-    
+
+    @Column
     private String name;
+
+    @Column
     private int age;
+
+    @Column
+    @Enumerated(EnumType.STRING)
     private AnimalType type;
+
+    @Column
+    @Enumerated(EnumType.STRING)
     private AnimalGender gender;
+
+    @Column
+    @Enumerated(EnumType.STRING)
     private AnimalHealthStatus healthStatus;
+
+    public Animal() {
+    }
+
+    public Animal(String name, int age, AnimalType type, AnimalGender gender, AnimalHealthStatus healthStatus) {
+        this.name = name;
+        this.age = age;
+        this.type = type;
+        this.gender = gender;
+        this.healthStatus = healthStatus;
+    }
+
+    public Animal(Long id, String name, int age, AnimalType type, AnimalGender gender, AnimalHealthStatus healthStatus) {
+        this.id = id;
+        this.name = name;
+        this.age = age;
+        this.type = type;
+        this.gender = gender;
+        this.healthStatus = healthStatus;
+    }
```

Comment:
Check if those can be done via Lombok - at least partially. Also, why do we need two different constructors?

Reply:
I have added Lombok annotations for noArgsConstructor and allArgsConstructor. The third one was not needed.